sway-core/src/semantic_analysis/ast_node/declaration/abi.rs

```diff
@@ -75,6 +85,16 @@ impl TypedAbiDeclaration {
             warnings,
             errors
         );
+        for typed_fn in &methods {
+            for param in &typed_fn.parameters {
+                if param.is_reference {
```

Comment:
Same as above.

Basically, I think we should avoid the assumption that `ref` => `mut` or vice versa, even though that's the only thing we support today.